js/src/visitor/toarray.ts

```diff
@@ -88,18 +88,6 @@ export class ToArrayVisitor extends Visitor {}
 
 /** @ignore */
 function arrayOfVector<T extends DataType>(vector: VectorType<T>): T['TArray'] {
-
-    const { type, length, stride } = vector;
-
-    // Fast case, return subarray if possible
-    switch (type.typeId) {
```

Comment:
Instead of removing the fast case can you just add a check to skip this if the type is nullable? As you pointed out in the jira we could still do this for non-nullable arrays.